six/common/cgo_free.c

```diff
@@ -16,6 +16,10 @@ void _set_cgo_free_cb(cb_cgo_free_t cb) {
 // free method to release memory allocated in the agent once we're done with
 // them.
 void cgo_free(void *ptr) {
+    // Technically this is not thread-safe as `cb_cgo_free` assignment
```

Comment:
Callbacks have to be initialized before initializing Python and therefore using `cgo_free`. In that way we're guaranteed that we won't have any issue with threading.

Reply:
Yup, I'm fine with the current code as-is, just wanted to add a note in case we ever made changes to this - which is highly unlikely anyways.